df_auth/serializers.py

```diff
@@ -92,3 +100,31 @@ def validate(self, attrs):
                 backend_module().generate_challenge(**attrs, **self.context)
 
         return attrs
+
+
+class AbstractSocialIdentitySerializer(serializers.Serializer):
+    user = None
+    provider = serializers.CharField(write_only=True)
+    code = serializers.CharField(write_only=True)
+
+
+class SocialTokenObtainSerializer(
+    AbstractTokenSerializer, AbstractSocialIdentitySerializer
+):
+    def validate(self, attrs):
+        request = self.context["request"]
+        user = request.user
```

Comment:
this should not be needed at all after you split theserializer into TokenObtain and Connect

Reply:
I created a specific serializer for SocialConnect and did some changes to them. 👍